internal-dns/src/dropshot_server.rs

```diff
@@ -21,17 +21,18 @@ impl Context {
 pub fn api() -> dropshot::ApiDescription<Arc<Context>> {
     let mut api = dropshot::ApiDescription::new();
 
-    api.register(dns_records_get).expect("register dns_records_get");
```

Comment:
@smklein I probably should have broken the non-lint changes in this file into its own PR, but if these look overwhelmingly uncontroversial I'll leave them in. Please let me know